lib/FrameManager.js

```diff
@@ -291,55 +291,93 @@ class Frame {
   }
 
   /**
-   * @param {string} filePath
-   * @return {!Promise<*>}
+   * @param {Object} options
+   * @return {!Promise}
    */
-  async injectFile(filePath) {
-    let contents = await new Promise((resolve, reject) => {
-      fs.readFile(filePath, 'utf8', (err, data) => {
-        if (err) return reject(err);
-        resolve(data);
+  async addScriptTag(options) {
+    if (typeof options.url === 'string')
+      return this.evaluate(addScriptUrl, options.url);
+
+    if (typeof options.path === 'string') {
+      let contents = await new Promise((resolve, reject) => {
+        fs.readFile(options.path, 'utf8', (err, data) => {
```

Comment:
Let me merge in #879 so that we can reuse the readFileAsync here.

Reply:
Good idea!!

Reply:
@SamVerschueren ok it's in. Please, make a readFileAsync with `helper.promisify` and use it here.